lib/rspec_profiling/collectors/statsd.rb

```diff
@@ -0,0 +1,100 @@
+require "statsd";
+require "logger";
+require "open3";
+
+
+
+module RspecProfiling
+  module Collectors
+    class Statsd
+      #delegate :timing, :increment, :count, :gauge, :time, :batch, to: :statsd, allow_nil: true 
+      NAMESPACE = 'rspec_profiling'
+      #Property used by CSV Collector, leaving it here
+      #to maintain reference of available properties.
+      HEADERS = %w{
+        branch
+        commit_hash
+        date
+        file
+        line_number
+        description
+        status
+        exception
+        time
+        query_count
+        query_time
+        request_count
+        request_time
+      }
+
+      attr_accessor :statsd, :results, :logger
+      
+      def self.install
+        new.install
+      end
+
+      def self.uninstall
+        new.uninstall
+      end
+
+      def health(host, port)
+        `nc -vuzn #{host} #{port} 2>&1 > /dev/null`.include? 'open'
+      end
+
+      def healthCheck
+        if(self.health RspecProfiling.config.statsd_host, RspecProfiling.config.statsd_port) 
+          @logger.info('Green light for statsd client')
+        else
+          @logger.warn('RED LIGHT for statsd client')
+        end
+      end
+
+      def self.reset
+        @results = Array.new;
+      end
+
+      def install
+        self.statsd.increment 'profiling_live'
+      end
+
+      def uninstall
+        self.statsd.decrement 'profiling_live'
+      end
+
+
+      def initialize
+        RspecProfiling.config.statsd_host ||= '127.0.0.1';
+        RspecProfiling.config.statsd_port ||= '8125';
+        @results = Array.new;
+        @resultType = Struct.new(:description, :process_time)
+        @statsd = false
+        @logger = Logger.new(STDOUT)
+      end
+
+      def statsd
+        if(!@statsd)
```

Comment:
code style: no parentheses around `if` condition. Proper way for null check is:
`if @statsd.nil?`

Reply:
Also, early returns are always clearer. Idiomatic method structure should be:
```
def statsd
  return @statsd unless @statsd.nil?
  self.health_check
  @statsd = ::Statsd.new(...)
end

Reply:
should it be `if @statsd.nil?()` ?

Reply:
No, parameter-less method calls don't need parentheses.

The reason for discouraging parameterized method calls without parentheses is because it leads to really difficult to find bugs in nested calling scenarios. For example, consider if you have something like this:
```
def foo(x, y)
  puts "#{x}, #{y}"
  x + y
end

def bar(a, b)
  puts "#{a}, #{b}"
  a * b
end

puts "#{foo 1, bar 2, 3}"
puts "#{bar 1, 2 - foo 4, 3}"
```

What does this output? Readability and debugging nightmare, right?